### PR TITLE
Problem: fire-cli argument parsing logic is unexpected

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- switch to click package

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

## Summary by CodeRabbit

- **New Features**
	- Introduced a revamped command-line interface using `click`, enhancing command invocation and parameter handling.
	- Added new commands for generating configurations and managing images, improving usability.
- **Bug Fixes**
	- Streamlined command execution flow, ensuring better interaction with the application.
- **Chores**
	- Updated project dependencies, removing `fire` and adding `click`.

### CODE REVIEW

File: testground/benchmark/benchmark/stateless.py [hunk 1]
```diff
@@ -38,149 +38,193 @@
...
+@click.option("--hostname-template", default=HOSTNAME_TEMPLATE)
+@click.option("--validators", default=3)
+@click.option("--fullnodes", default=7)
+@click.option("--num-accounts", default=10)
+@click.option("--num-txs", default=1000)
+@click.option("--config-patch", default="{}", callback=validate_json)
+@click.option("--app-patch", default="{}", callback=validate_json)
+@click.option("--genesis-patch", default="{}", callback=validate_json)
+@click.option("--validator-generate-load/--no-validator-generate-load", default=True)
+def gen(**kwargs):
```
Comment (mmsqe) [inline testground/benchmark/benchmark/stateless.py:64]: seems can't wrap whole options as before?

Comment (yihuang) [reply testground/benchmark/benchmark/stateless.py:64]: the `generic-gen` command do that.

Comment (yihuang) [reply testground/benchmark/benchmark/stateless.py:64]: and it also covers all the options including `outdir` and `hostname-template`.

Comment (mmsqe) [reply testground/benchmark/benchmark/stateless.py:64]: yea, might need modify readme later
```
nix run .#stateless-testcase generic-gen '{
  "outdir": "/tmp/outputs",
  "validators": 1,
  "fullnodes": 1,
  "num_accounts": 2,
  "num_txs": 1,
  "app_patch": {
    "mempool.max-txs": -1
  },
  "config_patch": {
    "mempool.size": 30000
  }
}'
```

**Review by mmsqe — APPROVED** [review]

**Pull request merged**